test/integration/client-side-encryption/driver.test.ts

```diff
@@ -1242,4 +1244,136 @@ describe('CSOT', function () {
+    };
+    const schemaMap = {
+      [dataNamespace]: {
+        bsonType: 'object',
+        properties: {
+          encrypted_placeholder: {
+            encrypt: {
+              keyId: '/placeholder',
+              bsonType: 'string',
+              algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Random'
+            }
+          }
+        }
+      }
+    };
+    let secureContextOptions;
+
+    before(async function () {
+      const caFile = await fs.readFile(process.env.CSFLE_TLS_CA_FILE);
+      const certFile = await fs.readFile(process.env.CSFLE_TLS_CLIENT_CERT_FILE);
+      secureContextOptions = {
+        ca: caFile,
+        key: certFile,
+        cert: certFile
+      };
+    });
+
+    context('when no driver specific TLS options are provided', function () {
+      let client;
+      let clientEncryption;
+      const options = {
+        keyVaultNamespace,
+        kmsProviders: { aws: getCSFLEKMSProviders().aws },
+        tlsOptions: {
+          aws: {
+            secureContext: tls.createSecureContext(secureContextOptions)
+          }
+        },
+        extraOptions: getEncryptExtraOptions()
+      };
+
+      beforeEach(async function () {
+        client = this.configuration.newClient({}, { autoEncryption: { ...options, schemaMap } });
+        clientEncryption = new ClientEncryption(client, options);
+        await client.connect();
+      });
+
+      afterEach(async function () {
+        await client.db(keyVaultDbName).collection(keyVaultCollName).deleteMany();
+        await client.close();
+      });
+
+      it('succeeds to connect', async function () {
+        // Use client encryption to create a data key. If this succeeds, then TLS worked.
+        const awsDatakeyId = await clientEncryption.createDataKey('aws', {
+          masterKey,
+          keyAltNames: ['aws_altname']
+        });
+        expect(awsDatakeyId).to.have.property('sub_type', 4);
+        // Use the client to get the data key. If this succeeds, then the TLS connection
+        // for auto encryption worked.
+        const results = await client
+          .db(keyVaultDbName)
+          .collection(keyVaultCollName)
+          .find({ _id: awsDatakeyId })
+          .toArray();
+        expect(results)
+          .to.have.a.lengthOf(1)
+          .and.to.have.nested.property('0.masterKey.provider', 'aws');
+      });
+    });
+
+    context('when driver specific TLS options are provided', function () {
+      let client;
+      let clientEncryption;
+      // Note we set tlsCAFile and tlsCertificateKeyFile to 'nofilename' to also
+      // test that the driver does not attempt to read these files in this case.
+      const options = {
+        keyVaultNamespace,
+        kmsProviders: { aws: getCSFLEKMSProviders().aws },
+        tlsOptions: {
+          aws: {
+            secureContext: tls.createSecureContext(secureContextOptions),
+            tlsCAFile: 'nofilename',
+            tlsCertificateKeyFile: 'nofilename'
```

Comment:
This covers 2 test bullet points in the AC with one test:
- This tests the secureContext overrides the tls* options
- This tests we don't attempt to read the tls* options when they are overridden. If that had happened, this test would throw.